extension/src/shared/ui-components/FeedBackModal.tsx

```diff
@@ -0,0 +1,210 @@
+import { FormControl } from "@mui/base/FormControl";
+import { Input, inputClasses } from "@mui/base/Input";
+import { TextareaAutosize } from "@mui/base/TextareaAutosize";
+import CancelIcon from "@mui/icons-material/Cancel";
+import {
+  Box,
+  Button,
+  Checkbox,
+  FormControlLabel,
+  IconButton,
+  Link,
+  Modal,
+  Typography,
+  styled,
+} from "@mui/material";
+import { red } from "@mui/material/colors";
+import { useCallback, useMemo, useState } from "react";
+
+const style = {
+  position: "absolute" as const,
+  top: "50%",
+  left: "50%",
+  transform: "translate(-50%, -50%)",
+  width: 420,
+  bgcolor: "background.paper",
+  color: "#000",
+  margin: "auto",
+  padding: "18px 22px",
+};
+
+export type Props = {
+  show: boolean;
+  setShowFeedbackModal: (show: boolean) => void;
+  onSubmit: (params: {
+    name: string;
+    email: string;
+    comment: string;
+    attachMapReview: boolean;
+  }) => void;
+};
+
+const FeedBackModal: React.FC<Props> = ({ show, setShowFeedbackModal, onSubmit }) => {
+  const [name, setName] = useState("");
+  const [email, setEmail] = useState("");
+  const [comment, setComment] = useState("");
+  const [attachMapReview, setAttachMapReview] = useState(false);
+
+  const handleFeedBackModalForm = useCallback(() => {
+    setName("");
+    setEmail("");
+    setComment("");
+    setAttachMapReview(false);
+    setShowFeedbackModal(false);
+  }, [setShowFeedbackModal]);
+
+  const handleSubmit = async () => {
+    const params = { name, email, comment, attachMapReview };
+    try {
+      await onSubmit(params);
+      handleFeedBackModalForm();
+    } catch {
+      // if any Error occur;
+    }
+  };
+
+  const handleSetName = useCallback((value: string) => {
+    setName(value);
+  }, []);
+
+  const handleSetEmail = useCallback((value: string) => {
+    setEmail(value);
+  }, []);
+
+  const handleSetComment = useCallback((value: string) => {
+    setComment(value);
+  }, []);
+
+  const handleSetAttachedMap = useCallback((value: boolean) => {
+    setAttachMapReview(value);
+  }, []);
+
+  const disabled = useMemo(() => {
+    if (comment && email) return false;
+    return true;
+  }, [comment, email]);
+
+  return (
+    <Modal
+      open={show}
+      aria-labelledby="modal-modal-title"
+      aria-describedby="modal-modal-description">
+      <Box sx={style}>
+        <Typography id="modal-modal-title" variant="h6" component="h2">
+          フィードバック{" "}
+          <IconButton
```

Comment:
Seems the style of this close button is different with design.
I guess the `variant` props should be like `variant="text"`